examples/testsuite-overwrites/readme.md

```diff
@@ -190,7 +190,7 @@ file: examples/testsuite-overwrites/portman-config.crm.json >>
       "overwrite": true
     },{
       "key": "monetary_amount",
-      "value": "{{$randomInt}}",
+      "value": "{{{$randomInt}}}",
```

Comment:
$randomInt is a [Postman dynamic variable](https://learning.postman.com/docs/tests-and-scripts/write-scripts/variables-list/). 

For the example case, my suggestion would be to use a regular defined Postman variable and not a dynamic variable. Otherwise it might come across that Dynamic Variables needs {{{}}}, which is not the case.

Reply:
That's a good suggestion @thim81  In the dynamic test scripts readme, I set up a use case where I captured the previous value of monetary_amount and then ran a pre-request script that doubled it before setting it in the PATCH request. So, in that example, I did use a regular Postman variable.

I struggled to develop an example of when you would want to set a number property in a request body in the CRM API, without doing some math on the monetary_value.  Do you have any ideas of other number properties that could logically be set to something findable in a Postman variable?   

Are there any examples of setting a Postman variable in a .env file?   Would it make sense to update this example to include a .env where that had something like:

PORTMAN_NEW_MONETARY_VALUE=150000

Or would that unnecessarily add complexity to this example of how to use them?

Reply:
A example with a variable using .ENV or globals. collectionVariables would be simplest. 

I would not show calculations, since you aim to showcase the variable usage.

Reply:
For your info, I'm planning to release a new version in the coming days, so ideally this PR can land and it will be included.

Reply:
I went back in and looked at this, and I'm not sure I understand your objection anymore.  At this point, we are building a request body to create a new lead:

```json
{
  "openApiOperationId": "leadsAdd",
  "overwriteRequestBody": [
    {
      "key": "name",
      "value": "{{$randomName}}",
      "overwrite": false
    },
    {
      "key": "company_name",
      "value": "{{$randomCompanyName}} {{$randomColor}}",
      "overwrite": true
    },{
      "key": "monetary_amount",
      "value": "{{{$randomInt}}}",
      "overwrite": true
    },
    {
      "key": "websites[0]",
      "remove": true
    },
    {
      "key": "social_links[1].url",
      "remove": true
    }
  ]
}
```

The example makes extensive use of Postman dynamic variables.   For example right before we set the monetary_amount property to $random_int, we set the name and company_name properties using dynamic variables as well.   The only place in the example were we are using the 3 curly brace syntax though is when we want the value not to be interpreted as a string.   This is true for dynamic variables as well as for regular collection variables.   

Net/Net  I don't think this example implies that dynamic variables need the 3 curly brace syntax.  Rather it tries to describe that you need to use the 3 curly brace syntax anytime you want the resulting generated postman response to not treat the value of that variable as a string.

I suspect that adding variables set in the environment would only complicate this example.

I hope that makes sense.  Please let me know if I have somehow missed your point.

Reply:
the comment was not about the usage of the Postman dynamic variables but the combination of the 3x {{{}}} since it is not needed. 

Example:
<img width="833" alt="image" src="https://github.com/user-attachments/assets/19d41222-6101-48dc-a7b2-bd06bbfba37b">

Results in a Postman request without quotes and which will become a number (not string) in the actual request
<img width="1404" alt="image" src="https://github.com/user-attachments/assets/b7d81b12-f214-440f-9db9-2d697e726bb8">

So my suggestion was to use a regular variable instead with 3x{{{}}}}, if you want to showcase the conversion of a number value OR keep the  Postman dynamic variables but only with 2x{{}}

Reply:
Oh, I see.  That is interesting that the dynamic variables are "smart" enough to know when they need to be quoted, but collection variables aren't.   I gather that is the Portman engine with those smarts? (yes, I should probably just look at the code).    It makes sense that the Portman engine doesn't have any advanced knowledge of the type of variable for a collection of variables, so it defaults to string.

However, it needs to be clarified for the end user since sometimes they need three curly braces, and sometimes they don't.  I think the documentation now needs to say something like the following:

"When using the `overwriteRequestBody` directive to assign Postman dynamic variables to request body properties, portman understands which dynamic variables are strings and will generate a request body that includes quotes around the variable name, while leaving references to dynamic variables which are numbers, booleans, or arrays, unquoted.

"When working with collection variables, Portman does not know the underlying variable type and defaults to treating the variable as a string value, and quoting the variable name in the generated request body.  If you know that your request body expects a number, boolean, or array for the property, encapsulate the collection variable name with three curly braces instead of 2."

Does that make sense to you?  Is it too much information?  (I certainly only understood this distinction once we had this dialog.

I must say that as I wrote this, it did occur to me that while Portman does not know the underlying type of the variable, it MAY know the underlying type of the request body property as defined in the spec and could be smart about when to use quotes, but that may be a feature request beyond the scope of this discussion.

Reply:
You are correct. Portman detects specific "$random" variables that will return a number value, if not it will treat it as a string.
For the regular variables, Portman cannot know, so we introduced the triple {{{}}} to provide the option for the user to instruct Portman to handle it as a number.

Your explanation is correct but the essence might get lost, perhaps a simple bullet list or little table would be sufficient?
- Use Postman Dynamic variable as a string: "{{$randomCompanyName}}"
- Use Postman Dynamic variable as a number: "{{$randomInt}}"
- Use Postman variables as a string: "{{$yourVar}}"
- Use Postman variables as number: "{{{$yourNumberVar}}}"

Portman could know the expected type if it is set in OpenAPI, but that would prevent overwriting it with a different type in case you're adding "unhappy" contract tests. For example, when you would expect the API to fail when a number value is entered for a string. 

We kept it flexible (and tried to keep it simple) to give the freedom to the users.

Reply:
FYI: We really appreciate your effort and input, so hope you keep going, even with this kind feedback 👍

Reply:
This kind of feedback is great.   I'm learning more about how Portman works with each iteration.   I'm about to go on vacation with limited computer access in a few hours, though.  

I'll try to update this PR before I leave.   Feel free to edit directly if it's still not on target, or I can pick it up again, even if I miss your upcoming release.

Reply:
Enjoy the vacation 🏖️